src/types/bytearray.rs

```diff
@@ -59,19 +97,35 @@ impl PyByteArray {
             std::ptr::write_bytes(buffer, 0u8, len);
             // (Further) Initialise the bytearray in init
             // If init returns an Err, pypybytearray will automatically deallocate the buffer
-            init(std::slice::from_raw_parts_mut(buffer, len)).map(|_| pypybytearray.into_ref(py))
+            init(std::slice::from_raw_parts_mut(buffer, len)).map(|_| pypybytearray.into_bound(py))
```

Comment:
I'd suggest going slightly further here and changing `pypybytearray` to be `Bound<PyByteArray>` already on line 93. 

It's a tiny detail, but in the error case it's more efficient to `Drop` a `Bound` than a `Py`, because for a `Py` it has to check if the GIL is held during drop. `Bound` already knows the GIL is held.

Reply:
Sounds good to me, I changed it to create a `Bound` directly and used its `as_ptr()` to extract the buffer pointer.